src/main/kotlin/sliv/tool/importer/ProjectParser.kt

```diff
@@ -2,42 +2,77 @@ package sliv.tool.importer
+    private val frames = FXCollections.observableArrayList<ProjectFrame>()
 
-    val files = FXCollections.observableHashMap<Long, MutableList<OutputInfo>>()
-    val images = FXCollections.observableArrayList<ImageInfo>()
-
-    fun createTreeWithFiles(path: String?, tree: TreeItem<String>): TreeItem<String> {
+    fun parseDirectory(path: String): Project {
         if (path == null) {
-            return tree
+            throw Exception("Directory not selected")
         }
         var directory = File(path)
 
         for (folder in directory.listFiles()) {
-            if (folder.name == IMAGE_DIRECTORY_NAME) {
-                images.addAll(folder.listFiles().map {
-                    ImageInfo(it.nameWithoutExtension, it.absolutePath)
-                })
-            } else {
+            if (folder.name != IMAGE_DIRECTORY_NAME) {
+                var splitName = folder.name.split("_")
+                var name = splitName[0]
+                var kind = splitName[1]
+                when (kind) {
+                    "keypoint" -> layers.add(ProjectLayer(LayerKind.KEYPOINT, name))
+                    "plane" -> layers.add(ProjectLayer(LayerKind.PLANE, name))
+                    "line" -> layers.add(ProjectLayer(LayerKind.LINE, name))
+                }
                 folder.listFiles().map {
-                    var key = it.nameWithoutExtension.toLong()
-                    if (files.containsKey(key)) {
-                        files[key]?.add(OutputInfo(folder.nameWithoutExtension, it.absolutePath))
+                    if (landmarks.containsKey(it.nameWithoutExtension.toLong())) {
+                        landmarks[it.nameWithoutExtension.toLong()]?.add(
+                            LandmarkFile(
+                                layers.last(),
+                                Path(it.path),
+                                extractUIDs(it.path)
+                            )
+                        )
                     } else {
-                        files[key] = mutableListOf(OutputInfo(folder.nameWithoutExtension, it.absolutePath))
+                        landmarks.put(
+                            it.nameWithoutExtension.toLong(),
+                            mutableListOf(LandmarkFile(layers.last(), Path(it.path), extractUIDs(it.path)))
+                        )
                     }
                 }
+            } else {
+                folder.listFiles().map {
+                    images.put(it.nameWithoutExtension.toLong(), it.absolutePath)
+                }
             }
         }
+        landmarks.forEach { (name, outputs) ->
+            frames.add(ProjectFrame(name, Path(images[name].toString()), outputs))
+        }
+        return Project(Path(path!!), frames, layers)
+    }
+
+    fun createTreeWithFiles(project: Project, tree: TreeItem<String>): TreeItem<String> {
+        val imgIcon = Image("file:icons/photo.png")
+        val fileIcon = Image("file:icons/description.png")
```

Comment:
Usually .svg files is used for such icons

Reply:
.svg files is not displayed(

Reply:
May be something like that [library](https://github.com/hervegirod/fxsvgimage)